source/src/main/java/com/kezong/fataar/RTransform.java

```diff
@@ -0,0 +1,202 @@
+        }
+
+        final File outputDir = outputProvider.getContentLocation("classes", getOutputTypes(), getScopes(), Format.DIRECTORY);
+
+        try {
+            for (final TransformInput input : transformInvocation.getInputs()) {
+                for (final DirectoryInput directoryInput : input.getDirectoryInputs()) {
+                    final File directoryFile = directoryInput.getFile();
+
+                    final ClassPool classPool = new ClassPool();
+                    classPool.insertClassPath(directoryFile.getAbsolutePath());
+
+                    for (final File originalClassFile : getChangedClassesList(directoryInput)) {
+                        if (!originalClassFile.getPath().endsWith(".class")) {
+                            continue; // ignore anything that is not class file
+                        }
+
+                        Future<?> submit = executor.submit(() -> {
+                            try {
+                                File relative = FilesKt.relativeTo(originalClassFile, directoryFile);
+                                String className = filePathToClassname(relative);
+                                final CtClass ctClass = classPool.get(className);
+                                ClassFile classFile = ctClass.getClassFile();
+                                ConstPool constPool = classFile.getConstPool();
+                                constPool.renameClass(transformTable);
+                                ctClass.writeFile(outputDir.getAbsolutePath());
+                            } catch (CannotCompileException | NotFoundException | IOException e) {
+                                e.printStackTrace();
+                            }
+                        });
+
+                        futures.add(submit);
+                    }
+                }
+            }
+        } catch (NotFoundException e) {
+            throw new RuntimeException(e);
+        }
+
+
+        for (Future<?> future : futures) {
+            try {
+                future.get();
+            } catch (ExecutionException e) {
+                e.printStackTrace();
+            }
+        }
+
+        futures.clear();
+
+        long endTime = System.currentTimeMillis();
+        System.out.println("[fat-aar]Task :transformClassesWithRenameRFor"
+                + transformInvocation.getContext().getVariantName()
+                + " cost "
+                + (endTime - startTime)
+                + "ms");
+    }
+
+    private Map<String, String> buildTransformTable(String variantName) {
+        final List<String> resourceTypes = Arrays.asList("anim", "animator", "array", "attr", "bool", "color", "dimen",
+                "drawable", "font", "fraction", "id", "integer", "interpolator", "layout", "menu", "mipmap", "plurals",
+                "raw", "string", "style", "styleable", "transition", "xml");
+
+        Collection<AndroidArchiveLibrary> libraries = VariantDependenciesStore.getLibraries(variantName);
```

Comment:
I think using static to pass properties from plugin to task is a bad practice - better to use Gradle's Property API

Reply:
I haven't found a good way to get the package name of the submodule. Do you have any suggestions?

Reply:
I don't see any other way, but to parse AndroidManifest.xml of submodule. Plugin can pass `Set<File>` through property, where each file corresponds to submodule manifest, and then `RTransform` can parse manifests for actual package names inside `transform()` - that way no additional I/O would be done during configuration time.